python_modules/dagster-pipes/dagster_pipes/__init__.py

```diff
@@ -181,6 +182,43 @@ class PipesException(TypedDict):
 # ##### UTIL
 # ########################
 
+ESCAPE_CHARACTER = "\\"
+
+
+def de_escape_asset_key(asset_key: str) -> str:
+    r"""Removes the backward slashes escape characters from the asset key.
+
+    Example: "foo\\/bar" -> "foo/bar"
+    """
+    return asset_key.replace(ESCAPE_CHARACTER, "")
```

Comment:
This will delete _any_ backslash, which we don't want since backslash is also (currently) an allowed character in asset keys. What we want to do is to replace "\/" with "/" and "\\" with "\":

AssetKey -> escaped str -> unescaped str

`AssetKey(["foo/bar"])` -> "foo\/bar" -> "foo/bar"
`AssetKey(["foo/bar", "baz"])` -> "foo\/bar/baz" -> "foo/bar/baz"
`AssetKey(["foo\bar"])` -> "foo\\bar" -> "foo\bar"